router/handle.go

```diff
@@ -579,7 +585,7 @@ func (rt *Handle) findWorkerSlot(ctx context.Context, workers []*worker, job *jo
 		return nil, types.ErrBarrierExists
 	}
 	rt.logger.Debugf("EventOrder: job %d of orderKey %s is allowed to be processed", job.JobID, orderKey)
-	if rt.shouldThrottle(ctx, job, parameters) {
+	if rt.shouldThrottle(ctx, job, parameters) && !abortedJob {
```

Comment:
Can't we check if we should throttle before entering the barrier like we do for backoff?

Something like:
1. if shouldBackoff AND !abortedJob THEN return ErrJobBackoff
2. if shouldThrottle AND !abortedJob THEN return ErrDestinationThrottled
3. get worker slot
4. enter barrier THEN return (ErrBarrierExists OR slot)

Wdyt?

Reply:
I'm not sure if this order(`barrier.Enter` and then checking if `throttled`) is being followed for a reason(can't think of one right now).
@atzoum wdyt?